### ISSUE TITLE
WCAG 4.1.3 AA: Carousel region not marked up (Issue A6)

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#352]
### Summary

Global template issue – The home page contains a group of slides and controls that are visually and functionally a carousel. These elements have not been enclosed in a labelled landmark region.

Via Vision Australia assessment: August 2024

Link text to download the Digital Service Standard document and poster contains file name information and characters.

The links also contain a title attribute that duplicates this text, which results in the link name being announced twice by some screen readers.

### Steps to reproduce

digital.gov.au beta | digital.gov.au (https://www.digital.gov.au) or any civic theme website where slider component is used.

![Picture1](https://github.com/user-attachments/assets/b619490c-47a4-4690-a2bd-68f6aad27dca)

### Why this matters

Landmark regions are a common navigational shortcut used by screen reader users, which can assist in quickly locating page sections and content. They also provide important semantic information on page structure and the purpose of content. 
When carousels are not marked up in a labelled region, screen reader users may not be aware of the carousel, or understand the meaning and purpose of its controls. 

```
<div class="ct-slider__container" data-slider="">
<h3 class="ct-heading ct-theme-light ct-slider__title">
Featured initiatives
</h3>
<div class="ct-slider__slides" data-slider-panel="" style="height: 428px;">
<div class="ct-slider__slides__inner" data-slider-rail="" style="width: 2496px; height: 428px;">
<div class="ct-slide ct-theme-light ct-slide--image-left " data-slider-slide="" data-slider-slide-hidden="true" style="width: 1248px; left: 0px; height: 428px;">
<div class="ct-slide__image">
<img class="ct-image ct-theme-light " src="/sites/default/files/images/2024-08/ai-policy-cover_square.jpg" alt="An image of the front cover of the Australian Government's policy for the responsible use of AI in government.">
</div>
<div class="ct-slide__wrapper">
<div class="ct-slide__inner">
<h2 class="ct-heading ct-theme-light ct-slide__title">
AI in government policy
</h2>
<div class="ct-paragraph ct-theme-light ct-paragraph--regular ct-slide__content">
<p>
<span dir="ltr">
Policy and standards supporting the Australian Government's use of artificial intelligence.
</span>
</p>
</div>
… 
</div>
</div>
</div>
… 
</div>
</div>
</div>

```

### Recommendation

Preferably, consider the need for a carousel component. Carousels can present multiple barriers for people with disabilities and increase cognitive load. As the amount of content in this slider is very small, it could be presented in another way, such as Call to Action (CTA) cards or as a list. 

Alternatively, create a landmark region for the carousel that encompasses all of its content and controls:

Wrap the carousel in a <div> element with the role="region" 

- Provide an accessible name for the region via the aria-labelledby attribute, referencing the "Featured initiatives" heading that precedes the carousel.
- Add the aria-roledescription="carousel" to the <div> element so that assistive technologies identify the region as a carousel.

Each slide should be in a <div> element with role="group"
- Provide an accessible name for each slide group via the aria-label attribute. The value should describe the position of the slide within the group, for example "1 of 2".
- Add the aria-roledescription="slide" to the <div> element so that assistive technologies identify the group as a slide.

### Code example

```
<div role="region" aria-roledescription="carousel" aria-labelledby="carousel-heading">
<h3 id="carousel-heading">
Featured initiatives
</h3>
… 
<!-- Slides and controls -->
<div role="group" aria-roledescription="slide" aria-label="1 of 2">
…
</div>
<button type="button">
Previous slide
</button>
<button type="button">
Next slide
</button>
</div>
```
Note: For further information regarding carousels, please refer to the Carousel (Slide Show or Image Rotator) Pattern | APG | WAI | W3C: https://www.w3.org/WAI/ARIA/apg/patterns/carousel/ 

### Related issues

For other issues related to this component, please refer to:

•	Issue 16 – Loss of carousel content when text resized
•	Issue 18 – Carousel functionality lost when reflowed
•	Issue 25 – Visually hidden carousel elements in mobile focus order
•	Issue 26 – Focus lost when carousel controls are disabled

**Comment by fionamorrison23** [comment]
Hi @CJE001 would you be able to provide the priority here as you have done on other tickets? Thanks.
